app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,34 @@
+class SessionsController < ApplicationController
+  before_action :find_user, only: [:create]
+
+  def new; end
+
+  def create
+    if @user&.authenticate(params.dig(:session, :password))
+      # Log the user in and redirect to the user's show page:
+      reset_session
+      log_in @user
+       # Assuming a 'show' action for User model
+      redirect_to @user, status: :see_other
+    else
+      flash.now[:danger] = t("invalid_email_password_combination")
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    log_out
+    redirect_to root_url, status: :see_other
+  end
+
+  private
+
+  def find_user
+    @user = User.find_by(email: params.dig(:session, :email)&.downcase)
+    if @user.nil?
+      flash.now[:danger] = t("user_not_found")
+      render :new, status: :unprocessable_entity
+    end
```

Comment:
Viết if như thế này cũng được nhưng thường để clear code hơn vì viết kiểu return if nha
```suggestion
    return if @user
    
    flash.now[:danger] = t("user_not_found")
    render :new, status: :unprocessable_entity
```